src/test/java/com/puppycrawl/tools/checkstyle/AbstractModuleTestSupport.java

```diff
@@ -137,39 +94,42 @@ protected ModuleCreationOption findModuleCreationOption(String moduleName) {
      */
     protected final Checker createChecker(Configuration moduleConfig)
             throws Exception {
-        final String name = moduleConfig.getName();
+        final String moduleName = moduleConfig.getName();
+        final Checker checker = new Checker();
+        checker.setModuleClassLoader(Thread.currentThread().getContextClassLoader());
+
+        if (ROOT_MODULE_NAME.equals(moduleName)) {
+            checker.configure(moduleConfig);
+        }
+        else {
+            final Class<?> moduleClass = Class.forName(moduleName);
 
-        return createChecker(moduleConfig, findModuleCreationOption(name));
+            addConfigToChecker(checker, moduleClass, moduleConfig);
+        }
+
+        checker.addListener(getBriefUtLogger());
+        return checker;
     }
 
     /**
-     * Creates {@link Checker} instance based on the given {@link Configuration} instance.
+     * Identifies the way to add the {@code moduleConfig} to the {@code checker} instance.
      *
+     * @param checker {@link Checker} instance.
+     * @param moduleClass {@link Class} of the module involved.
      * @param moduleConfig {@link Configuration} instance.
-     * @param moduleCreationOption {@code IN_TREEWALKER} if the {@code moduleConfig} should be added
-     *                                                  under {@link TreeWalker}.
-     * @return {@link Checker} instance based on the given {@link Configuration} instance.
-     * @throws Exception if an exception occurs during checker configuration.
+     * @throws Exception if an exception occurs during configuration.
      */
-    protected final Checker createChecker(Configuration moduleConfig,
-                                 ModuleCreationOption moduleCreationOption)
-            throws Exception {
-        final Checker checker = new Checker();
-        checker.setModuleClassLoader(Thread.currentThread().getContextClassLoader());
-
-        if (moduleCreationOption == ModuleCreationOption.IN_TREEWALKER) {
+    protected void addConfigToChecker(Checker checker, Class<?> moduleClass,
```

Comment:
`addConfigToItsParent(Checker, Configuration)`

moduleClass is already in moduleConfig.

Reply:
> addConfigToItsParent
> module does not go directly to Checker

The method is named as such because the end result it is trying to add `moduleConfig` to `Checker` but there could be something intermediary between the 2, such as `TreeWalker`. If `TreeWalker` is the parent, then the suggested method name is misleading as we are doing more than adding it to `TreeWalker`. What we really need is to add it to Checker.

So saying "it does not go directly to Checker" is wrong.

> (Checker, Configuration)
> moduleClass is already in moduleConfig.

This was to simplify this method for overloading. The `Class` is needed to pass to `ModuleReflectionUtil` to know it's heritage. To remove it from the method, means the code creating the class has to be moved in this method, and any overrides of this method will need to duplicate it as the `Class` is always needed.

Also `moduleConfig` only has the class in the Test tier. IT tier works with the configuration and thus does not already have the class in `moduleConfig`. This is what is preventing synchronization right now. When IT tier can be synchronized fully to test, it will need a special method to convert config to class, and thus it makes this method more interoperable that it receive the class instead of creating it itself, as IT will have need for a custom conversion.

Reply:
this method actually do "configure" so it is instantiation of modules , lets call it `configureChecker`.

```
protected final Checker createChecker(Configuration moduleConfig)
            throws Exception {
        final String moduleName = moduleConfig.getName();
        final Checker checker = new Checker();
        checker.setModuleClassLoader(Thread.currentThread().getContextClassLoader());

        if (ROOT_MODULE_NAME.equals(moduleName)) {
            checker.configure(moduleConfig);
        }
        else { 
           configureChecker(checker, moduleConfig);
        }

        checker.addListener(getBriefUtLogger());
        return checker;
    }

    protected void configureChecker(Checker checker, Configuration moduleConfig) throws Exception {
          final Class<?> moduleClass = Class.forName(moduleName);

        if (ModuleReflectionUtil.isCheckstyleTreeWalkerCheck(moduleClass)
                || ModuleReflectionUtil.isTreeWalkerFilterModule(moduleClass)) {
            final Configuration config = createTreeWalkerConfig(moduleConfig);
            checker.configure(config);
        }
        else {
            final Configuration config = createRootConfig(moduleConfig);
            checker.configure(config);
        }
    }
```

Reply:
Done.

Reply:
Not done.
`configureChecker(checker, moduleClass, moduleConfig);`

Should be
`configureChecker(checker, moduleConfig);`

As I showed in my code above.

Reply:
See comment at https://github.com/checkstyle/checkstyle/pull/11603#discussion_r867400591

Reply:
I pushed the change.